lib/Models/Subscriber.php

```diff
@@ -108,7 +108,8 @@ function isWooCommerceUser() {
   function getLinkToken() {
     if ($this->link_token === null) {
       $this->link_token = self::generateToken($this->email);
-      $this->save();
+      // `$this->save()` fails if the subscriber has subscriptions, segments or custom fields
```

Comment:
Having a `$user` instance, if you call `$user->withSubscriptions()` it will add a `subscriptions` attribute to `$user`. Then calling `$user->save()` tries to save that attribute to the database causing an error. My initial plan was to change the `save()` method to unset the `subscriptions` attribute, save the model then set it again. Hence the commit message. But I quickly learned that there are other attributes added by `withSegments` and `withCustomFields` making it harder to unset all of them. So my quickest fix was to use an UPDATE query. What do you think?

Reply:
Ok, this is really 🤦🏻‍♂️crazy behaviour. I wasn't aware of that. I hope we will start using Doctrine soon. (I know we won't 😭)